core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -706,10 +709,27 @@ protected File getRootDirFor(String name) {
      * {@link #getActiveInstance()} provides the checked versions of the method.
      * @return The instance. Null if the {@link Jenkins} instance has not been started,
      * or was already shut down
+     * @since 1.653
      */
-    @CLIResolver
     @CheckForNull
+    public static Jenkins getInstanceOrNull() {
+        return HOLDER.getInstance();
+    }
+
+    /**
+     * Gets the {@link Jenkins} singleton. In certain rare cases you may have code that is intended to run before
+     * Jenkins starts or while Jenkins is being shut-down. For those rare cases use {@link #getInstanceOrNull()}.
```

Comment:
Not so much that the code is _intended_ to run while Jenkins is being shut down, as that it runs asynchronously, and `cleanUp()` is not exhaustive, so on occasion it could run after shutdown, until you find a way to ensure that it does not.